data/mongodb/v0/config/tasks.json

```diff
@@ -0,0 +1,457 @@
+    "input": {
+      "instillUIOrder": 0,
+      "properties": {
+        "database-name": {
+          "description": "Fill in your MongoDB database name",
+          "instillUpstreamTypes": [
+            "reference",
+            "template",
+            "value"
+          ],
+          "instillAcceptFormats": [
+            "string"
+          ],
+          "instillUIOrder": 0,
+          "title": "Database Name",
+          "type": "string"
+        },
+        "collection-name": {
+          "description": "The name of the collection in the MongoDB database",
+          "instillAcceptFormats": [
+            "string"
+          ],
+          "instillShortDescription": "Collection Name",
+          "instillUIOrder": 1,
+          "instillUpstreamTypes": [
+            "reference",
+            "template",
+            "value"
+          ],
+          "title": "Collection Name",
+          "type": "string"
+        },
+        "data": {
+          "description": "The data to be inserted",
+          "instillAcceptFormats": [
+            "semi-structured/*","structured/*","object"
+          ],
+          "instillShortDescription": "JSON Data",
+          "instillUIOrder": 2,
+          "instillUpstreamTypes": [
+            "reference",
+            "template",
+            "value"
+          ],
+          "title": "Data"
+        }
+      },
+      "required": [
+        "database-name",
+        "collection-name",
+        "data"
+      ],
+      "title": "Input",
+      "type": "object"
+    },
+    "output": {
+      "instillUIOrder": 0,
+      "properties": {
+        "status": {
+          "description": "Insert status",
+          "instillFormat": "string",
+          "required": [],
+          "instillUIOrder": 0,
+          "title": "Status",
+          "type": "string"
+        }
+      },
+      "required": [
+        "status"
+      ],
+      "title": "Output",
+      "type": "object"
+    }
+  },
+  "TASK_FIND": {
+    "instillShortDescription": "Perform a find operation",
+    "input": {
+      "instillUIOrder": 0,
+      "properties": {
+        "database-name": {
+          "description": "Fill in your MongoDB database name",
+          "instillUpstreamTypes": [
+            "reference",
+            "template",
+            "value"
+          ],
+          "instillAcceptFormats": [
+            "string"
+          ],
+          "instillUIOrder": 0,
+          "title": "Database Name",
+          "type": "string"
+        },
+        "collection-name": {
+          "description": "The name of the collection in the MongoDB database",
+          "instillAcceptFormats": [
+            "string"
+          ],
+          "instillShortDescription": "Collection Name",
+          "instillUIOrder": 1,
+          "instillUpstreamTypes": [
+            "reference",
+            "template",
+            "value"
+          ],
+          "title": "Collection Name",
+          "type": "string"
+        },
+        "criteria": {
```

Comment:
> Question Does it make sense to call it filter? [source](https://www.mongodb.com/docs/manual/tutorial/query-documents/) ![image](https://private-user-images.githubusercontent.com/64747455/350442258-b7b19c1f-e747-4ab5-9bef-25cd2f625eee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0MDUwNjEsIm5iZiI6MTcyMTQwNDc2MSwicGF0aCI6Ii82NDc0NzQ1NS8zNTA0NDIyNTgtYjdiMTljMWYtZTc0Ny00YWI1LTliZWYtMjVjZDJmNjI1ZWVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE5VDE1NTkyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZjNTMwN2E4MjcyZDE1MmQ4YWRiOTk3OWZiZWM0MjcwNTk2NzY4YzA2MGIyNjgzMWUxZTJjN2Y2MDc3NDE1MmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5DVAA5vQr07cgNPJq2t6LVNgAeVQnpBvYtp9jtclZYw)

Yes, I feel like filter sounds better

Reply:
@chuang8511

Yes, I also feel that "filter" fits better than "criteria." Will change it soon!

Regarding the use of SQL commands with NoSQL databases, in my opinion, it’s better to use the corresponding database standard and language. Some people may only be familiar with MongoDB, which could be why they choose it and then SQL-like structure input comes out of nowhere doesn't seem to be a good UX for me. In this case, MongoDB uses "aggregate" for filtering and takes JSON or struct as its input.

Additionally, using SQL queries directly may harm the database such as "SQL Injection." For example, if we have a query like "SELECT * FROM inventory WHERE {user input}," the user input could be manipulated into something like "username = '' OR 1=1; DROP TABLE inventory;" which can delete all the data in the table inventory. This is the trade off I have been considering, even when deciding to use JSON for SQL component rather than SQL query. Please your advice on this matter.

[sql injection](https://medium.com/@craftingcode/understanding-sql-injection-attacks-and-how-to-prevent-them-8750ff1384fd)

Reply:
> In this case, MongoDB uses "aggregate" for filtering and takes JSON or struct as its input.

OK, I do not have much experience with MongoDB. If aggregate makes more sense in community, we should use it!